pkg/secrets/vault_test.go

```diff
@@ -35,3 +40,127 @@ func TestValidateVaultConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestNoSecret(t *testing.T) {
+	notASecret := "notASecret"
+	eng, err := NewDecrypter(context.TODO(), notASecret)
+	assert.Nil(t, err)
+
+	unchanged, _ := eng.Decrypt()
+	assert.Equal(t, notASecret, unchanged)
+}
+
+func TestNoVaultConfig(t *testing.T) {
+	e := Engines["vault"]
+	delete(Engines, "vault")
+	decrypter, err := NewDecrypter(context.TODO(), "encrypted:vault!e:secret!n:test-secret!k:foo")
+	assert.NotNil(t, err)
+	assert.Nil(t, decrypter)
+	Engines["vault"] = e
+}
+
+func TestClientGetsNamespace(t *testing.T) {
+	cases := map[string]struct {
+		cfg VaultConfig
+	}{
+		"with namespace": {
+			cfg: VaultConfig{
+				Enabled:    true,
+				Url:        "http://127.0.0.1:8200",
+				AuthMethod: "KUBERNETES",
+				Namespace:  "my-namespace",
+				Role:       "my-role",
+				Path:       "my-path",
+				Token:      "my-token",
+			},
+		},
+		"no namespace": {
+			cfg: VaultConfig{
+				Enabled:    true,
+				Url:        "http://127.0.0.1:8200",
+				AuthMethod: "KUBERNETES",
+				Namespace:  "",
+				Role:       "my-role",
+				Path:       "my-path",
+				Token:      "my-other-token",
+			},
+		},
+	}
+
+	for _, c := range cases {
+		d := VaultDecrypter{
+			vaultConfig: c.cfg,
+		}
+
+		client, err := d.FetchVaultClient(c.cfg.Token)
+		assert.Nil(t, err)
+
+		header := client.Headers()
+		if header != nil {
+			assert.Equal(t, c.cfg.Namespace, header.Get(consts.NamespaceHeaderName))
+		}
+
+		assert.Equal(t, c.cfg.Token, client.Token())
+	}
+}
+
+func TestParseVaultSecret(t *testing.T) {
+	cases := map[string]struct {
+		params      string
+		expected    func(*testing.T, *VaultDecrypter)
+		shouldError bool
+	}{
+		"standard syntax": {
+			"e:engine!p:path!k:key",
+			func(t *testing.T, decrypter *VaultDecrypter) {
+				assert.Equal(t, "engine", decrypter.engine)
+				assert.Equal(t, "path", decrypter.path)
+				assert.Equal(t, "key", decrypter.key)
+			},
```

Comment:
I realize you just refactored what was already here, so no need to address this as part of this PR. 

First, I appreciate the use of `map[string]struct` :D So much more clear as to what the test is testing.

I'm guilty of doing this but usually when I see table driven tests that define a test function as part of the test case, it's usually a smell that something could be different. Ideally, the test case defines a set of inputs and the test should be generic enough to test all cases.

The easiest way to refactor this might be to replace the `expected` function with `expectedDecryptor`, defining the value you and then comparing those using `assert.EqualValues`

Reply:
yep, feels cleaner 👍